Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift

```diff
@@ -453,10 +467,26 @@ open class MessageComposerViewController<ExtraData: ExtraDataTypes>: ViewControl
     public func textViewDidChange(_ textView: UITextView) {
         isEmpty = textView.text.replacingOccurrences(of: " ", with: "").isEmpty
         replaceTextWithSlashCommandViewIfNeeded()
-        promptSuggestionIfNeeded(for: textView.text)
+
+        updateMentionFlag(with: textView.text as NSString, till: textView.selectedRange.location)
+
+        promptSuggestionIfNeeded(for: textView.text!)
         composerView.invalidateIntrinsicContentSize()
     }
 
+    func updateMentionFlag(with text: NSString, till caret: Int) {
```

Comment:
`NSString` ?? 🦕

Reply:
`NSRange` and `Range<String.Index>` don't like each other and do not cooperate too well with different kind of strings. I feel this can be done better, in the same time I don't know how -> I spent already too much time figuring out how to do this with `String`, then @ManWithBear and prove me you can sometimes use something with `NS` prefix and it doesn't hurt...

Reply:
In 2 words: 
- `String` and `NSString` have different unicode handling so you cannot interchange indexes between them
- `UITextView` uses `NSAttributedString` subclass for text storage (`NSTextStorage`), which operates on `NSString`